numismatic/feeds/poloniex.py

```diff
@@ -59,68 +92,61 @@ def _handle_trade(msg, subscription):
-                    id=trade_msg[1],) #  id comes from the original message
-
-
-        subscription.event_stream.emit(event_msg)
+
+        if msg[2][1][0] == 't':
+            raw_data = msg[2][1]
+
+            event = Trade(exchange=subscription.exchange,
+                        symbol=subscription.symbol, 
+                        price=raw_data[3],
+                        volume=raw_data[4],
+                        type='SELL' if raw_data[2] == 0 else 'BUY',
+                        timestamp=raw_data[5],
+                        id=raw_data[1],) #  id comes from the original message
+
+
+            subscription.event_stream.emit(event)
     
     @staticmethod
-    def _handle_orderbook(msg, subscription):
+    def _handle_orderbook_removemodify(msg, subscription):
         '''
             Poloniex order book format:
              [148,394056638,[["o",0,"0.07615527","0.34317849"]]]
              which is [currency pair, id, o (orderbook), 0/1 (remove/modify)
              price, quantity]
         '''
-        if len(msg) != 3:
-            return
-        if len(msg[2]) != 2:
-            return
-
-        # Only interested in order book messages
-        if msg[2][0][0] != 'o':
-           return
-
-        orderbook_msg = msg[2][0]
-
-        event_msg = Order(
-            exchange=subscription.exchange,
-            symbol=subscription.symbol,
-            price=orderbook_msg[2],
-            volume=float(orderbook_msg[3]),
-            # The orderbook type can either be modify or remove
-            # for modify, the amount will replace any previous
-            # value. I am using buy because I can't find anyway
-            # to distuingish and the documentation uses bid
-            type='CANCEL' if orderbook_msg[1] == 0 else 'BUY',
-            id=msg[1],
-        )
-
-        subscription.event_stream.emit(event_msg)
+        if msg[2][0][0] == 'o':
+            raw_data = msg[2][0]
+            if raw_data[3] == '0.00000000':
+                event = Order(
+                    exchange=subscription.exchange,
+                    symbol=subscription.symbol,
+                    price=raw_data[2],
+                    volume=raw_data[3],
+                    type='CANCEL',
+                )
+            else:
+                # Below is an OrderModify,
+                # but represented as an order
+                event = Order(
+                    exchange=subscription.exchange,
+                    symbol=subscription.symbol,
+                    price=raw_data[2],
+                    volume=raw_data[3],
+                    type='ASK' if raw_data[1] == 0 else 'BID',
+                )
+
+            subscription.event_stream.emit(event)
 
 class PoloniexFeed(Feed):
-
     _websocket_client_class = PoloniexWebsocketClient
   
     @staticmethod
     def get_symbol(asset, currency):
-        return f'{asset}-{currency}'
+        # Poloniex requires format USDT_<currency symbol>
```

Comment:
I wonder why they have that convention?

Reply:
Poloniex in general is weird. Zero documentation - thank goodness I found a Node project and I was able to copy what they do (in terms of their format).